### PR TITLE
add 101 symmetric-tree

### LINKED ISSUE #41: add 101 Symmetric Tree
## Summary
add 101

## Goal
add 101

## Todo
- [x] add 101
- [x] write time and space complexity

## Deadline
4/24

## Parent issue
None

## References
None

## Notes
None

### PR DESCRIPTION
**Pull request by kobakobashu**
time : 50m

### CODE REVIEW

**Comment by kobakobashu** [comment]
Please check my code.

File: 101.symmetric-tree.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        if not left and not right:
+            return True
+        if not left or not right:
+            return False
+        
+        if left.val == right.val:
+            outtree = self.isMirror(left.left, right.right)
+            intree = self.isMirror(left.right, right.left)
+            return outtree and intree
+        else:
```
Comment (ryuji0123) [inline 101.symmetric-tree.py:34]: You don't need to write else. Just return

**Review by ryuji0123 — CHANGES_REQUESTED** [review]
Good algorithm

**Comment by kobakobashu** [comment]
I fixed my code.  Please check it.

File: 101.symmetric-tree.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    def isSymmetric(self, root: TreeNode) -> bool:
+        """
+        time: O(N)  N : the number of all nodes
+        space: O(N)
+        """
+        if not root:
+            return True
+        else:
+            return self.isMirror(root.left, root.right)
+    def isMirror(self, left, right):
```
Comment (ryuji0123) [inline 101.symmetric-tree.py:24]: insert a line before function definition.

**Review by ryuji0123 — CHANGES_REQUESTED** [review]

**Comment by kobakobashu** [comment]
I fixed my code.  Please check my code.

File: 101.symmetric-tree.py (hunk 2 above)
Comment (ryuji0123) [reply 101.symmetric-tree.py:25]: Check function name's character in PEP8. Unfortunately, function names in Leetcode break it, so you cannot use the same name style without checking coding style guides.

**Review by ryuji0123 — CHANGES_REQUESTED** [review]

**Comment by kobakobashu** [comment]
I rename function name.  Please check it.

File: 101.symmetric-tree.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+#         self.right = right
+class Solution:
+    def isSymmetric(self, root: TreeNode) -> bool:
+        """
+        time: O(N)  N : the number of all nodes
+        space: O(N)
+        """
+        if not root:
+            return True
+        else:
```
Comment (ryuji0123) [inline 101.symmetric-tree.py:22]: remove else or write like this:

```py
return True if not root else self.check_mirror(root.left, root.right)
```

**Review by ryuji0123 — CHANGES_REQUESTED** [review]

**Comment by kobakobashu** [comment]
I removed else and wrote type hints and docstring.  Please check it.

**Pull request merged**